### PR TITLE
fix: graphql pagination, query editor css, editor constants

### LINKED ISSUE #15714: [Bug]: Evaluated Values of Graphql Query, Pagination is always undefined
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

The query body, pagination params should evaluate to proper values without dynamic bindings or undefined.

Screenshot 2022-08-04 at 6 29 31 PM

### Steps To Reproduce

- Try creating a graphql api
- Change the query body
- The evaluated popup has undefined as value always

### Public Sample App

_No response_

### Version

Self Hosted

### LINKED ISSUE #16094: [Bug]: Pagination settings dropdown has some wrong caching
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

In [this](http://3.111.33.176/app/graphql-basic/page1-62f9dc05241ba712c63b43e4/edit/api/62fa03fa241ba712c63b43ee) example app, I had (once upon a time) declared a variable with name `lim`. I have since deleted this variable.

Now when I go to the `Pagination` tab of my query, I see that `lim` is pre-selected, and I am also unable to expand the dropdowns. 

Screenshots:
1. This query does not have any variables: 
Screenshot 2022-08-17 at 13 45 31

2. But when I go to Pagination tab, I see some old redundant variable from another query being shown here. Also, I am unable to expand the dropdown:
Screenshot 2022-08-17 at 13 46 23

### Steps To Reproduce

explained above

### Public Sample App

_No response_

### Version

Cloud

### LINKED ISSUE #16182: [Bug]: GraphQL: fix background color
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

```
The visual design of the query editor  in body seems inconsistent. Specifically the grey BG behind ‘Query’ and ‘Query Variables’. Screenshot attached.
```
Ref: https://theappsmith.slack.com/archives/CGBPVEJ5C/p1655885054803919?thread_ts=1655821041.009059&cid=CGBPVEJ5C

### Steps To Reproduce

https://theappsmith.slack.com/archives/CGBPVEJ5C/p1655885054803919?thread_ts=1655821041.009059&cid=CGBPVEJ5C

### Public Sample App

_No response_

### Version

self hosted

### LINKED ISSUE #16183: [Bug]: GraphQL: Incorrect bracket pairing during linting
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

```
In the query window in body tab, the linting of {{ }} is wrong, and I’m not sure why it’s different vs. other editors. If I add double curly braces, the linter assumes that bracket 1 and 3 are a pair and bracket 2 and 4 are a pair.
```
Ref: https://theappsmith.slack.com/archives/CGBPVEJ5C/p1655885054803919?thread_ts=1655821041.009059&cid=CGBPVEJ5C

### Steps To Reproduce

https://theappsmith.slack.com/archives/CGBPVEJ5C/p1655885054803919?thread_ts=1655821041.009059&cid=CGBPVEJ5C

### Public Sample App

_No response_

### Version

feature branch DP

### PR DESCRIPTION
**Pull request by AmanAgarwal041**
## Description

- Updated the graphql parser to parse variables for pagination dropdowns
- Fixed the isSeparate flag for Next Limit Value and Previous Limit Value
- Graphql Query Editor CSS
- Evaluated value for pagination variables with binding paths 
- Separated Editor Constants for Api and Graphql
- Re-arranged datasources icons for creation of graphql api, authenticated graphql
- Highlighting brackets for query syntax

Fixes #16182
Fixes #15714
Fixes #16094
Fixes #16183

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?

> Please describe the tests that you ran to verify your changes. Provide instructions, so we can reproduce.
> Please also list any relevant details for your test configuration.

- Test A
- Test B

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: app/client/src/components/editorComponents/ApiResponseView.tsx [hunk 1]
```diff
@@ -37,7 +37,7 @@ import EntityBottomTabs from "./EntityBottomTabs";
 import { DEBUGGER_TAB_KEYS } from "./Debugger/helpers";
 import { setCurrentTab } from "actions/debuggerActions";
 import Table from "pages/Editor/QueryEditor/Table";
-import { API_RESPONSE_TYPE_OPTIONS } from "constants/ApiEditorConstants";
+import { API_RESPONSE_TYPE_OPTIONS } from "constants/ApiEditorConstants/CommonConstants";
```
Comment (ApekshaBhosale) [inline app/client/src/components/editorComponents/ApiResponseView.tsx:40]: can you give a different name, please? this indicates anything common but these are common API type constants, isn't it?

File: app/client/src/utils/editor/EditorBindingPaths.ts [hunk 2]
```diff
@@ -0,0 +1,50 @@
+// import {
+//   LIMITBASED_PREFIX,
+//   CURSORBASED_PREFIX,
```
Comment (ApekshaBhosale) [inline app/client/src/utils/editor/EditorBindingPaths.ts:3]: remove this code

File: app/client/src/utils/editor/EditorBindingPaths.ts [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+//   CURSOR_NEXT_PREFIX,
+//   CURSOR_PREVIOUS_PREFIX,
+// } from "constants/ApiEditorConstants";
+import { PaginationSubComponent } from "components/formControls/utils";
+import { EvaluationSubstitutionType } from "entities/DataTree/dataTreeFactory";
+
+// Graphql Pagination Constants
+const LIMITBASED_PREFIX = "limitBased";
+const CURSORBASED_PREFIX = "cursorBased";
```
Comment (ApekshaBhosale) [inline app/client/src/utils/editor/EditorBindingPaths.ts:12]: import it from GraphQL constants please

File: app/client/src/sagas/ApiPaneSagas.ts [hunk 4]
```diff
@@ -560,12 +565,16 @@ function* handleCreateNewApiActionSaga(
   const workspaceId: string = yield select(getCurrentWorkspaceId);
   const { pageId, apiType = REST_PLUGIN_PACKAGE_NAME } = action.payload;
   const pluginId: string = yield select(getPluginIdOfPackageName, apiType);
-  let defaultConfig: DEFAULT_CREATE_API_CONFIG_TYPE;
-  if (apiType in DEFAULT_CREATE_API_CONFIG) {
-    defaultConfig = DEFAULT_CREATE_API_CONFIG[apiType];
+  let defaultConfig;
+  if (apiType === REST_PLUGIN_PACKAGE_NAME) {
```
Comment (ApekshaBhosale) [inline app/client/src/sagas/ApiPaneSagas.ts:569]: shorten this please

**Review by ApekshaBhosale — CHANGES_REQUESTED** [review]

File: app/client/src/utils/editor/EditorBindingPaths.ts (hunk 3 above)
Comment (AmanAgarwal041) [reply app/client/src/utils/editor/EditorBindingPaths.ts]: Created constants in `EditorBindingPaths` and imported from this in `Graphql/Pagination` file.

**Pull request merged**